mozilla_django_oidc/auth.py

```diff
@@ -72,29 +72,60 @@ def create_user(self, claims):
 
         return self.UserModel.objects.create_user(username, email)
 
+    def get_algorithms(self):
+        """Returns the JWS verify algorithms"""
+        return ['HS256']
+
+    def get_verify_key(self):
+        """Returns the JWS verify key"""
+        return self.OIDC_RP_CLIENT_SECRET
+
     def verify_token(self, token, **kwargs):
         """Validate the token signature."""
+
         nonce = kwargs.get('nonce')
+        key = self.get_verify_key()
+        algorithms = self.get_algorithms()
 
         # Verify the token
-        verified_token = jws.verify(
-            token,
-            self.OIDC_RP_CLIENT_SECRET,
-            algorithms=['HS256']
-        )
+        verified_token = jws.verify(token, key, algorithms=algorithms)
         # The 'verified_token' will always be a byte string since it's
         # the result of base64.urlsafe_b64decode().
         # The payload is always the result of base64.urlsafe_b64decode().
         # In Python 3 and 2, that's always a byte string.
         # In Python3.6, the json.loads() function can accept a byte string
         # as it will automagically decode it to a unicode string before
         # deserializing https://bugs.python.org/issue17909
-        token_nonce = json.loads(verified_token.decode('utf-8')).get('nonce')
+        verified_id = json.loads(verified_token.decode('utf-8'))
+        token_nonce = verified_id.get('nonce')
 
-        if import_from_settings('OIDC_USE_NONCE', True) and nonce != token_nonce:
+        if import_from_settings('OIDC_USE_NONCE', True) \
+           and nonce != token_nonce:
             msg = 'JWT Nonce verification failed.'
             raise SuspiciousOperation(msg)
-        return True
+        return verified_id
+
+    def get_token_response(self, payload):
+        """Obtains the OIDC token"""
+
+        response = requests.post(
+            self.OIDC_OP_TOKEN_ENDPOINT,
+            data=payload,
+            verify=import_from_settings('OIDC_VERIFY_SSL', True))
+        response.raise_for_status()
+        return response.json()
+
+    def get_userinfo(self, access_token, id_token, verified_id):
```

Comment:
It doesn't look that we use ``id_token`` in this method. What's the purpose of having it there?

Reply:
The same applies for ``id_token`` as well.

Reply:
id_token is not used yet. but it may contain valuable info to use in get_userinfo()

Reply:
That fact needs to be documented with a comment. E.g. 
"""
The id_token is not used in the default implementation but it's here
in case someone wants to extend this class and use it in their get_userinfo.
"""
...or something better like that.

Reply:
done